### PR TITLE
Feature/configure sms bot from db

### PR DESCRIPTION
**Pull request by lewisdaly**
This is a pretty mammoth change, but surprisingly, it's cleared up a lot of the Thread, ConversationDelegate and ConversationRouter logic.

I've tested locally, but this might break the deployment, as we will need to re-seed the db etc.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ node_modules
 yarn.lock
 .mongo
 sms-utils/backups
+bundle.js
+bundle.js.map
```
Comment (kevindoveton) [inline .gitignore:7]: should the entire dist directory be ignored?

File: deployment/tag_and_push_docker.sh [hunk 2]
```diff
@@ -1,8 +1,9 @@
 #!/bin/bash
 
 echo 'Make sure you are logged in, with docker login'
-
-dirname=$(echo ${PWD##*/} | sed 's/-//g')
+#Get the dir 1 level up
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && cd ../ && pwd )"
```
Comment (kevindoveton) [inline deployment/tag_and_push_docker.sh:5]: you dont use this?

File: sms-bot/src/message/FacebookRouter.js [hunk 3]
```diff
@@ -47,35 +47,35 @@ class FacebookRouter {
         var data = req.body;
         if (data.object == 'page') {
           data.entry.forEach((entry) => {
-            
+
             entry.messaging.forEach((msg) => {
               console.log(msg)
```
Comment (kevindoveton) [inline sms-bot/src/message/FacebookRouter.js:52]: can probably remove the log // my bad

**Review by kevindoveton — CHANGES_REQUESTED** [review]
not sure why tests are failing - worked fine on my computer

File: .gitignore (hunk 1 above)
Comment (lewisdaly) [reply .gitignore:7]: Nah, the production running docker container actually looks for the index.html file in dist. Not ideal, but the best I could do after 30 mins of dealing with webpack

**Comment by kevindoveton** [comment]
facebook bot is broken, I think CLI is using a different validate params to the facebook router

**Comment by lewisdaly** [comment]
Ok. I'll check it out

**Pull request merged**